packages/core/src/crawlers/error_tracker.ts

```diff
@@ -0,0 +1,395 @@
+import { inspect } from 'node:util';
+
+import { ErrorSnapshotter } from './error_snapshotter';
+import type { CrawlingContext } from '../crawlers/crawler_commons';
+
+/**
+ * Node.js Error interface
+ */
+export interface ErrnoException extends Error {
+    errno?: number | undefined;
+    code?: string | number | undefined;
+    path?: string | undefined;
+    syscall?: string | undefined;
+    cause?: any;
```

Comment:
It was added there because of this:

```
this.add(error.cause);
```
this.add expects ErrnoException type and it shows an error, as `cause` type is unknown